### PR TITLE
feat: adding continuous delivery tab in git settings

### LINKED ISSUE #30510: CD: New tab for CD in CE
- Tab: Continuous Delivery behind a feature flag

### PR DESCRIPTION
**Pull request by brayn003**
## Description
Part of continuous delivery with git feature. Includes CE UI elements for "Continuous delivery" tab. Utilizes feature flag `release_git_continuous_delivery_enabled` to hide the implementation from UI

#### PR fixes following issue(s)
Fixes #30510 

#### Media
image

#### Type of change
- New feature (non-breaking change which adds functionality)\

- [ ] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/client/src/ce/components/gitComponents/GitSettingsCDTab/UnlicensedGitCD.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import { Button, Text } from "design-system";
+import { useAppsmithEnterpriseLink } from "pages/Editor/gitSync/GitSettingsModal/TabBranch/hooks";
+import React from "react";
+import styled from "styled-components";
+
+const Container = styled.div`
+  padding-top: 8px;
+  padding-bottom: 16px;
+  overflow: auto;
+  min-height: calc(360px + 52px);
```
Comment (sneha122) [inline app/client/src/ce/components/gitComponents/GitSettingsCDTab/UnlicensedGitCD.tsx:16]: @brayn003 Why are we doing computation here? Can't we simply use 412px?

File: app/client/src/ce/components/gitComponents/GitSettingsCDTab/UnlicensedGitCD.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+`;
+
+const SectionTitle = styled(Text)`
+  font-weight: 600;
```
Comment (sneha122) [inline app/client/src/ce/components/gitComponents/GitSettingsCDTab/UnlicensedGitCD.tsx:20]: Don't we have variation of Text component in design system that addresses this font weight?

Comment (brayn003) [reply app/client/src/ce/components/gitComponents/GitSettingsCDTab/UnlicensedGitCD.tsx:20]: There is but it doesn't support font-weight 600

File: app/client/src/ce/components/gitComponents/GitSettingsCDTab/UnlicensedGitCD.tsx (hunk 1 above)
Comment (brayn003) [reply app/client/src/ce/components/gitComponents/GitSettingsCDTab/UnlicensedGitCD.tsx:16]: I have done this for easier readability and maintenance. it means 360px modal height + 52px for footer. The computation is minimal and should not affect performance

File: app/client/src/ce/components/gitComponents/GitSettingsCDTab/UnlicensedGitCD.tsx (hunk 2 above)
Comment (sneha122) [reply app/client/src/ce/components/gitComponents/GitSettingsCDTab/UnlicensedGitCD.tsx:20]: Okay if it's required in multiple places, we can perhaps request design system team add config for this setting

**Review by sneha122 — APPROVED** [review]

**Pull request merged**